feat: Integrate JSX into snap notifications

hmalik88 opened the pull request:
## **Description**

Adds an expanded view for snaps notifications, using JSX content returned from the snap to populate the expanded view.


Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions:
1. What is the reason for the change? Allow for richer snaps notifications
2. What is the improvement/solution? Add expanded view for snaps, allowing a snap to return jsx content in the expanded view.

## **Screenshots/Recordings**


### **After**



https://github.com/user-attachments/assets/74c89bb7-7510-4d1c-acb6-a585978ecec8

## **Manual testing steps**
1. Pull down https://github.com/hmalik88/swiss-knife-snap and run `yarn start` to serve the snap and site.
2. Build this branch using `yarn start:flask`
3. Install the snap from the local host site.
4. Trigger a notification with the "trigger a JSX notification" button and observe the results.

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

hmalik88 committed:
make adjustments to show notification call

Mrtenz committed:
Bump Snaps packages

hmalik88 committed:
Merge branch 'develop' into hm/integrate-jsx-notifications

Mrtenz committed:
Bump dependencies again

hmalik88 committed:
update notification controller, update raw snap notification type

hmalik88 committed:
Merge branch 'develop' into hm/integrate-jsx-notifications

hmalik88 committed:
Merge branch 'mrtenz/bump-snaps-packages' into hm/integrate-jsx-notifications

hmalik88 committed:
add snap notification modal

hmalik88 committed:
Merge branch 'develop' into mrtenz/bump-snaps-packages

GuillaumeRx committed:
feat(snaps): Connect `getCurrencyRate` hook to `multichainRateController` (#27489)

hmalik88 committed:
feat: Integrate MetaMask links into Snaps `Link` component (#27377)

## **Description**


Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change? Previously, developers couldn't
navigate to parts of the extension with the existing functionality
provided to them through snaps UI
2. What is the improvement/solution? The Snaps `Link` component now
checks for if the url is a MetaMask url and navigates to the proper path
accordingly.


## **Screenshots/Recordings**

### **After**


## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

hmalik88 committed:
Merge branch 'mrtenz/bump-snaps-packages' into hm/integrate-jsx-notifications

hmalik88 committed:
Merge branch 'mrtenz/bump-snaps-packages' into hm/integrate-jsx-notifications

hmalik88 committed:
integrate snaps expanded view into notification system

hmalik88 committed:
update yarn lock

hmalik88 committed:
update lavamoat policies

hmalik88 commented on ui/pages/notifications/notification-components/index.ts:
```diff
@@ -9,9 +9,20 @@ import { components as StakeComponents } from './stake/stake';
 
-const { TRIGGER_TYPES } = NotificationServicesController.Constants;
-type TRIGGER_TYPES = NotificationServicesController.Constants.TRIGGER_TYPES;
+export enum SNAP_TRIGGER {
+  SNAP = 'snap',
+}
+
+export const TRIGGER_TYPES = {
+  ...NotificationServicesController.Constants.TRIGGER_TYPES,
+  ...SNAP_TRIGGER,
+};
+
+export type TRIGGER_TYPES =
+  | NotificationServicesController.Constants.TRIGGER_TYPES
+  | SNAP_TRIGGER;
```
Adjusting types here to accommodate for snaps, these will be adjusted upstream in a later PR.

hmalik88 committed:
lint fixes

hmalik88 committed:
another lint fix

hmalik88 committed:
more lint fixes

hmalik88 committed:
fix type issues

hmalik88 committed:
fix export

hmalik88 committed:
possible fix

hmalik88 committed:
use literal instead of enum

hmalik88 committed:
update notification detail button test

hmalik88 committed:
fix notification detail block explorer button test

hmalik88 committed:
update snap-utils

hmalik88 committed:
dedupe

hmalik88 committed:
Merge branch 'develop' into mrtenz/bump-snaps-packages

hmalik88 committed:
Merge branch 'mrtenz/bump-snaps-packages' into hm/integrate-jsx-notifications

hmalik88 committed:
update yarn.lock

hmalik88 committed:
Merge branch 'mrtenz/bump-snaps-packages' into hm/integrate-jsx-notifications

hmalik88 committed:
update yarn.lock

hmalik88 committed:
dedupe

hmalik88 committed:
Merge branch 'mrtenz/bump-snaps-packages' into hm/integrate-jsx-notifications

hmalik88 committed:
regenerate lock file

hmalik88 committed:
update policy

hmalik88 committed:
update policy

hmalik88 committed:
possible fix

hmalik88 committed:
revert changes

hmalik88 committed:
update policy final time?

Mrtenz committed:
Bump dependencies again again

Mrtenz commented:
@metamaskbot update-policies

Mrtenz commented on ui/components/multichain/notification-detail-button/notification-detail-button.tsx:
```diff
@@ -45,20 +58,25 @@ export const NotificationDetailButton = ({
     <Button
       key={id}
-      href={href}
+      {...(!isMetaMaskUrl && { href })}
```
Does this work?
```suggestion
      href={!isMetaMaskUrl && href}
```

Mrtenz commented on ui/contexts/metamask-notifications/metamask-notifications.tsx:
```diff
@@ -7,8 +7,11 @@ import { selectIsProfileSyncingEnabled } from '../../selectors/metamask-notifica
+import { SnapNotification } from '../../pages/notifications/snap/types/types';
 
-type Notification = NotificationServicesController.Types.INotification;
+type Notification =
```
Since this is used in several places, maybe extract it to a separate file.

Mrtenz commented on ui/pages/notification-details/notification-details-footer/notification-details-footer.tsx:
```diff
@@ -39,6 +39,11 @@ export const NotificationDetailsFooter = ({
         </>
       )}
+      {footer.type === 'footer_snap_notification' && (
+        <>
```
This fragment is unnecessary.

hmalik88 replied:
I thought so too, was just following the notification team's pattern 😅, I'll change it.

Mrtenz commented on ui/pages/notifications/notification-components/index.ts:
```diff
@@ -9,9 +9,20 @@ import { components as StakeComponents } from './stake/stake';
-const { TRIGGER_TYPES } = NotificationServicesController.Constants;
-type TRIGGER_TYPES = NotificationServicesController.Constants.TRIGGER_TYPES;
+export enum SNAP_TRIGGER {
+  SNAP = 'snap',
```
Should this be in the notification controller instead? 🤔

hmalik88 replied:
yeahh...but opening a PR in their repo would take longer. Doing this as a temp workaround.

Mrtenz commented on ui/pages/notifications/notification-components/snap/snap.tsx:
```diff
@@ -6,68 +6,93 @@ import {
+        },
+      });
+      history.push(getSnapRoute(notification.data.origin));
+    };
```
Nit:
```suggestion
    };

```

Mrtenz commented on ui/pages/notifications/notification-components/snap/snap.tsx:
```diff
@@ -6,68 +6,93 @@ import {
+  details: {
+    title: ({ notification }) => (
+      <NotificationDetailTitle
+        title={notification.data.expandedView?.title as string}
```
Is this property guaranteed to be set?

hmalik88 replied:
yes, this is only ever called when the notification route is hit

hmalik88 replied:
it is not

Mrtenz replied:
Maybe I misunderstand, but it's optional here:

https://github.com/MetaMask/snaps/blob/f4890dd4eaa1b318f6fad00aa456e4600bc97225/packages/snaps-sdk/src/types/methods/notify.ts#L40-L43

hmalik88 replied:
This component is only ever called in a notification detail route, the route is not navigated to unless there's a detailed view. The title is guaranteed to be there.

Mrtenz commented on ui/pages/notifications/notification-components/snap/snap.tsx:
```diff
@@ -6,68 +6,93 @@ import {
+        title={{
+          items: [
+            {
+              text: snapsNameGetter(notification.data.origin) || 'Snap',
```
Maybe use "Unknown Snap"? Is there a case in which this happens in the first place?
```suggestion
              text: snapsNameGetter(notification.data.origin) ?? 'Unknown Snap',
```

hmalik88 replied:
None that I know of, i think the notifications team just had it as a fail-safe. I'll change it.

Mrtenz commented on ui/pages/notifications/notification-components/snap/snap.tsx:
```diff
@@ -6,68 +6,93 @@ import {
+      Content: ({ notification }) => (
+        <SnapUIRenderer
+          snapId={notification.data.origin}
+          interfaceId={notification.data.expandedView?.interfaceId as string}
```
Is this property guaranteed to be set?

hmalik88 replied:
yup, see above.

Mrtenz commented on ui/pages/notifications/notification-components/snap/snap.tsx:
```diff
@@ -6,68 +6,93 @@ import {
+      notification.data.expandedView?.footerLink ? (
+        <NotificationDetailButton
+          notification={notification}
+          text={notification.data.expandedView?.footerLink.text as string}
+          href={notification.data.expandedView?.footerLink.href as string}
```
Are these properties guaranteed to be set?

hmalik88 replied:
Yes, see above.

Mrtenz commented on ui/pages/notifications/notification-components/types/notifications/notifications.ts:
```diff
@@ -32,6 +35,11 @@ type BodyFeatureAnnouncement<N = Notification> = {
 };
 
+type BodySnapNotification<N = Notification> = {
+  type: 'body_snap_notification';
```
Can you extract these into a separate type / enum?

Mrtenz commented on ui/pages/notifications/snap/types/types.ts:
```diff
@@ -1,8 +1,15 @@
-export const SNAP = 'SNAP' as const;
+import { SNAP_TRIGGER } from '../../notification-components';
+
+export type SnapNotificationExpandedView = {
```
Maybe this should be a type in `snaps-sdk`, so we can re-use it.

hmalik88 replied:
I can open an issue to add it in after and replace this.

Mrtenz commented on app/scripts/metamask-controller.js:
```diff
@@ -2756,14 +2756,19 @@ export default class MetamaskController extends EventEmitter {
             ),
-          showInAppNotification: (origin, args) =>
+          showInAppNotification: (origin, args) => {
+            const notificationArgs = { ...args };
+            notificationArgs.interfaceId = notificationArgs.content;
+            delete notificationArgs.content;
+            delete notificationArgs.type;
```
What's going on here?

hmalik88 replied:
Changing the shape to fit what the notification controller is expecting.

Mrtenz replied:
Can you just create a new object, rather than spreading and deleting properties afterwards?

Mrtenz committed:
Bump execution environment URL

hmalik88 committed:
Merge branch 'mrtenz/bump-snaps-packages' into hm/integrate-jsx-notifications

hmalik88 committed:
changes per PR comments

hmalik88 committed:
regenerate yarn.lock

hmalik88 committed:
make change per comment

hmalik88 committed:
fix

hmalik88 committed:
lint fix

hmalik88 committed:
possible import fix

hmalik88 committed:
another possible fix

FrederikBolding committed:
Fix circular dependency issues

FrederikBolding committed:
Fix another circular dependency

Mrtenz committed:
Merge branch 'develop' into mrtenz/bump-snaps-packages

GuillaumeRx committed:
feat(snaps): Add `size` prop to `Heading` (#27721)

## **Description

This PR adds support for the `size` prop to the `Heading` component.
Right now `md` is bind to `headingMd`, `lg` is bind to `headingLg` to
match
https://www.figma.com/design/2cqqTWJKoHcjYVGfNXxLbQ/Beyond-Ethereum?node-id=3458-5041&node-type=text&m=dev
as requested by @eriknson. The default behavior if `size` is not defined
si `headingSm`

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/27721?quickstart=1)

## **Related issues**

## **Manual testing steps**

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

hmalik88 committed:
lint fix

hmalik88 committed:
import fix

hmalik88 committed:
another fix

hmalik88 committed:
Merge branch 'mrtenz/bump-snaps-packages' into hm/integrate-jsx-notifications

hmalik88 committed:
add missing section

hmalik88 committed:
lint fix

hmalik88 committed:
lint fix

hmalik88 committed:
fix undefined error

FrederikBolding commented on app/scripts/metamask-controller.js:
```diff
@@ -2792,7 +2800,7 @@ export default class MetamaskController extends EventEmitter {
             return this.controllerMessenger.call(
               'PhishingController:testOrigin',
-              sender.url,
+              sender,
```
?

FrederikBolding replied:
Turns out this is a very real issue. Handling outside of this PR.

hmalik88 committed:
fix bug and make style changes

GuillaumeRx commented on ui/components/multichain/create-eth-account/create-eth-account.js:
```diff
@@ -7,7 +7,7 @@ import {
   getNextAvailableAccountName as getNextAvailableAccountNameFromController,
 } from '../../../store/actions';
-import { CreateAccount } from '..';
+import { CreateAccount } from '../create-account';
```
?

hmalik88 replied:
It was causing some test breaking, the import couldn't be read otherwise.

FrederikBolding commented on ui/contexts/metamask-notifications/metamask-notifications.tsx:
```diff
@@ -1,14 +1,12 @@
 import { getIsUnlocked } from '../../ducks/metamask/metamask';
-
-type Notification = NotificationServicesController.Types.INotification;
+import { type Notification } from '../../pages/notifications/notification-components/types/notifications/notifications';
```
Why are we changing these imports?

hmalik88 committed:
fix styling

hmalik88 committed:
Merge branch 'develop' into hm/integrate-jsx-notifications

hmalik88 committed:
regenerate yarn lock

hmalik88 committed:
Merge branch 'develop' into hm/integrate-jsx-notifications

hmalik88 committed:
Revert "Merge branch 'develop' into hm/integrate-jsx-notifications"

This reverts commit f9dcbf2205d84fc322fb221c17adc8493a640d36, reversing
changes made to 6ca3c8a291bfad1e88065a0f92bbf9adbd27df4f.

hmalik88 committed:
Revert "Revert "Merge branch 'develop' into hm/integrate-jsx-notifications""

This reverts commit ffc2c9e2bc754109cd086248f516b2787c8b0bcc.

hmalik88 committed:
undo

hmalik88 committed:
fix

hmalik88 committed:
Revert "undo"

This reverts commit 05e8d0e3bd71598bcc8018128cebb39b775f2972.

hmalik88 committed:
fix yarn lock

hmalik88 committed:
Merge branch 'develop' into hm/integrate-jsx-notifications

hmalik88 committed:
lint fix

hmalik88 committed:
fix read action in notification list item

hmalik88 committed:
remove extra hook

hmalik88 committed:
fix notification-details

hmalik88 committed:
lint fix

hmalik88 committed:
revert change to actions

hmalik88 committed:
update components

hmalik88 committed:
Merge branch 'develop' into hm/integrate-jsx-notifications

hmalik88 committed:
lint fix

hmalik88 committed:
lint fix

hmalik88 committed:
another lint fix

hmalik88 committed:
fixes

hmalik88 committed:
Merge branch 'develop' into hm/integrate-jsx-notifications

hmalik88 committed:
fix storybook

hmalik88 committed:
Merge branch 'develop' into hm/integrate-jsx-notifications

hmalik88 committed:
Merge branch 'develop' into hm/integrate-jsx-notifications

hmalik88 committed:
update logic to delete notification on detail view dismount

hmalik88 committed:
Merge branch 'develop' into hm/integrate-jsx-notifications

hmalik88 committed:
lint fix

FrederikBolding commented on ui/pages/notifications/notifications-list-item.tsx:
```diff
@@ -44,13 +46,23 @@ export function NotificationsListItem({
+      notification.type === TRIGGER_TYPES.SNAP &&
+      !hasProperty(notification.data, 'detailedView')
+    ) {
+      setNotificationTimeout(notification.id);
```
When is a notification with a detailed view removed?

hmalik88 replied:
On dismount

FrederikBolding commented on ui/pages/notifications/notification-components/stake/stake.tsx:
```diff
@@ -148,7 +151,7 @@ export const components: NotificationComponent<StakeNotification> = {
     },
     body: {
-      type: 'body_onchain_notification',
+      type: NotificationComponentType.OnChainBody,
```
In the future it would be nice to split this out in a separated PR and do this refactor before mixing in Snaps changes IMO

hmalik88 replied:
Yeah I suppose, did this on Maarten's request lol

FrederikBolding commented on ui/pages/notification-details/notification-details-footer/notification-details-footer.tsx:
```diff
@@ -29,21 +30,14 @@ export const NotificationDetailsFooter = ({
+      {footer.type === NotificationComponentType.AnnouncementFooter && (
+        <footer.ExtensionLink notification={notification} />
       )}
-      {footer.type === 'footer_feature_announcement' && (
```
Is this change intentional?

hmalik88 replied:
Yes

FrederikBolding replied:
I don't understand then, why are we removing elements from another type of notification?

hmalik88 replied:
oh shoot that was NOT intentional, good catch.

hmalik88 committed:
Merge branch 'main' into hm/integrate-jsx-notifications

FrederikBolding reviewed (commented):
I did not get all the way through yet, but some preliminary comments

hmalik88 committed:
ensure snap notification timeout hook runs on dismount

hmalik88 committed:
Merge branch 'main' into hm/integrate-jsx-notifications

hmalik88 commented on ui/pages/notifications/notifications-list-item.stories.tsx:
```diff
@@ -44,7 +44,12 @@ const notificationMocks = {
-  Snap: createMockSnapNotification,
+  Snap: () => {
+    const mock = createMockSnapNotification();
+    // TODO(hmalik88): the mock's origin should be fixed upstream
```
Not done, going to do it in a later PR

FrederikBolding commented on ui/components/multichain/notification-detail-button/notification-detail-button.tsx:
```diff
@@ -46,17 +56,21 @@ export const NotificationDetailButton = ({
+      href={!isMetaMaskUrl && href ? href : undefined}
       variant={variant}
-      externalLink={isExternal}
+      externalLink={!isMetaMaskUrl}
```
Seems like this would break existing uses of this component with `isExternal=false`?

hmalik88 replied:
Yeah, fixed.

FrederikBolding commented on ui/pages/notifications/notification-components/snap/snap.tsx:
```diff
@@ -1,98 +1,161 @@
+        title={{
+          items: [
+            {
+              text: snapsNameGetter(notification.data.origin) ?? 'Unknown Snap',
```
Is the else case even possible here?

hmalik88 replied:
Nah, remnants of old notifications team code. Removing it.

hmalik88 commented on app/scripts/metamask-controller.js:
```diff
@@ -1461,12 +1461,19 @@ export default class MetamaskController extends EventEmitter {
+
+            const detailedView = {
+              title,
+              ...(footerLink ? { footerLink } : {}),
```
I'd like to keep the object clean without `undefined`. The performance difference of creating the intermediate object is neglible.

FrederikBolding replied:
This is not a performance worry, but it adds additional complexity that makes it harder to reason about the code.

I'm not necessarily against it, but why do we need `undefined` to not be present here?

hmalik88 replied:
Technically it's not type correct if `footerLink` or `detailedView` is present on the notification object, also I have to do more type-fu to access the `detailedView` property instead of using `hasProperty` as I'm currently doing in some of the components.

GuillaumeRx replied:
I'll consider this as a nit that we can fix after if we really want. This would just improve code readability :)

hmalik88 commented on app/scripts/metamask-controller.js:
```diff
@@ -1461,12 +1461,19 @@ export default class MetamaskController extends EventEmitter {
               data: {
                 message,
                 origin,
+                ...(interfaceId ? { detailedView } : {}),
```
See above.

FrederikBolding commented on ui/pages/notifications/notification-components/snap/snap.tsx:
![image](https://github.com/user-attachments/assets/10ab9a8a-9167-4571-ac44-a8d151342d32)
This seems broken. 

Additionally:
- When using a link in a notification with an expanded view you cannot click the link anymore
- If you view an expanded view, go back and into the expanded view again the notification will disappear as you are reading it

hmalik88 replied:
1. Not using markdown so that's why, I'll fix that.
2. This is happening because on dismount the timeout is triggered, I don't have a timer that I'm using to pause if you decided to navigate back. I'll have to add that in if you think from a UX perspective it's better. Technically the view disappearing like this is correct...

hmalik88 replied:
> This seems broken.

Are you referring to the below points or something else?

FrederikBolding replied:
For 1. I was referring to when not in the expanded view, the link renders correctly but you cannot click it

hmalik88 replied:
That one is kind of tough to get around, I think snap developers might have to design the notification to account for if they provide a detailed view. It'll be better to provide a link inside the JSX content itself.

FrederikBolding replied:
Do you have an explanation for why the link isn't even clickable? Seems like we are letting some sort of click bubble up that shouldn't be?

hmalik88 replied:
Fixed here: [4ec8491](https://github.com/MetaMask/metamask-extension/pull/27407/commits/4ec84917629f56b5d2c30c5c19f2bf575036d626)

FrederikBolding commented on app/scripts/metamask-controller.js:
```diff
@@ -2922,14 +2929,22 @@ export default class MetamaskController extends EventEmitter {
+              title,
+              footerLink,
+            };
             this.controllerMessenger.call(
```
We need to return the response here otherwise every call to `snap_notify`  will throw
```
Request for method \"snap_notify\" returned undefined result."
```

hmalik88 committed:
revert footer change

hmalik88 committed:
use isExternal to determine if link is external in notification detail button

hmalik88 committed:
remove unnecessary else case

hmalik88 committed:
return snap_notify response

eriknson reviewed (dismissed):
Design lgtm!

hmalik88 committed:
address PR comments

hmalik88 committed:
Merge branch 'main' into hm/integrate-jsx-notifications

FrederikBolding commented on ui/components/app/snaps/snap-ui-markdown/snap-ui-markdown.js:
```diff
@@ -17,7 +17,7 @@ import {
 import SnapLinkWarning from '../snap-link-warning';
 
-const Paragraph = (props) => (
+export const Paragraph = (props) => (
```
Why is this exported now? Feels like an implementation detail for the component, especially with that name

hmalik88 replied:
Unintended, I was about to do something 😅

hmalik88 committed:
remove export

hmalik88 committed:
prevent modal rendering if not snap notification

hmalik88 committed:
Merge branch 'main' into hm/integrate-jsx-notifications

Prithpal-Sooriya commented on ui/pages/notification-details/notification-details.tsx:
```diff
@@ -59,7 +61,13 @@ function useEffectOnNotificationView(notificationData?: Notification) {
+        setNotificationTimeout(notificationData.id);
+      }
+    };
+  }, []);
```
Empty dep array is fine here. I will investigate our notifications on why we needed our effect for marking as read (I think we can remove it, but will do so in another PR)

Prithpal-Sooriya commented on ui/pages/notifications/notification-components/erc1155-sent-received/erc1155-sent-received.tsx:
```diff
@@ -189,7 +192,7 @@ export const components: NotificationComponent<ERC1155Notification> = {
   },
   footer: {
-    type: 'footer_onchain_notification',
+    type: NotificationComponentType.OnChainFooter,
```
Really appreciate this abstraction! I really like the exhaustive discrimination we can now do for OnChain, Feature Announcement, and now Snap Body/Footer/Etc.

Prithpal-Sooriya commented on ui/pages/notifications/notification-components/types/notifications/notifications.ts:
```diff
@@ -52,7 +71,13 @@ export type NotificationComponent<N extends Notification = Notification> = {
     title: NotificationFC<N>;
-    body: BodyFeatureAnnouncement<N> | BodyOnChainNotification<N>;
+    body:
+      | BodyFeatureAnnouncement<N>
+      | BodyOnChainNotification<N>
+      | BodySnapNotification<N>;
   };
-  footer: FooterFeatureAnnouncement<N> | FooterOnChainNotification<N>;
+  footer:
+    | FooterFeatureAnnouncement<N>
+    | FooterOnChainNotification<N>
+    | FooterSnapNotification<N>;
```
Nice. I think our team might do another refactor in the future to better group these unions.
E.g. it is unlikely that you would have a FeatureAnnonucement Body, but a Snap Footer.

IDK why we didn't group the footer in the `details` section 😛 

Noted and will work on this later.

hmalik88 committed:
make sure footer link displays warning modal

hmalik88 committed:
Merge branch 'main' into hm/integrate-jsx-notifications

hmalik88 committed:
make sure modal and external link is not shown for metamask links in markdown content

hmalik88 committed:
Merge branch 'main' into hm/integrate-jsx-notifications
